File: python/euklides2.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
# euklides2.py


def nwd_v2(a, b):
    while a > 0:
        a = a % b
        b = b - a
    return b


def main(args):
    a = int(input("Podaj liczbe naturalną: "))
    b = int(input("podaj liczbe naturalną: "))
    assert nwd_v2(5, 10) == 5
    assert nwd_v2(33, 11) == 11
    print("NWD({:d}, {:d}) = {:d}".format(a, b, nwd_v2(a, b)))

    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/euklides2.cpp
/*
 * euklides2.cpp
 */


#include <iostream>

using namespace std;

int nwd_v2(int a, int b)
{
    while ( a > 0)
    {
        a = a % b;
        b = b - a;
    }
    return b;
}

int nwd_rek(int a, int b)
{
	if (b == 0)
		return a;
	return nwd_rek(b, a % b);
}

int main(int argc, char **argv)
{
	int a, b;
	cout << "Podaj liczbe naturalną: " << endl;
	cin >> a;
	cout << "Podaj liczbe naturalną: " << endl;
	cin >> b;
	// cout << "Największy wspólny dzielnik: " << nwd_v2(a, b) << endl;
	printf("NWD(%d, %d) = %d", a, b, nwd_rek(a, b));	
	return 0;
}

